core/src/main/java/com/linecorp/armeria/client/BlockingConditionalResponseAs.java

```diff
@@ -0,0 +1,112 @@
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.ResponseEntity;
+
+public final class BlockingConditionalResponseAs<V>
+        extends ConditionalResponseAs<HttpResponse, AggregatedHttpResponse, ResponseEntity<V>> {
+    private static final Predicate<AggregatedHttpResponse> SUCCESS_PREDICATE = res -> res.status().isSuccess();
+
+    BlockingConditionalResponseAs(ResponseAs<HttpResponse, AggregatedHttpResponse> originalResponseAs,
+                                  ResponseAs<AggregatedHttpResponse, ResponseEntity<V>> responseAs,
+                                  Predicate<AggregatedHttpResponse> predicate) {
+        super(originalResponseAs, responseAs, predicate);
+    }
+
+    /**
+     * Invokes {@link ConditionalResponseAs#andThen(ResponseAs, Predicate)} to add the mapping of
+     * {@link ResponseAs} and {@link Predicate}.
+     */
+    public BlockingConditionalResponseAs<V> andThenJson(
+            Class<? extends V> clazz, Predicate<AggregatedHttpResponse> predicate) {
+        return (BlockingConditionalResponseAs<V>) andThen(
+                AggregatedResponseAs.json(clazz, predicate), predicate);
+    }
+
+    /**
+     * Invokes {@link ConditionalResponseAs#andThen(ResponseAs, Predicate)} to add the mapping of
+     * {@link ResponseAs} and {@link Predicate}.
+     */
+    public BlockingConditionalResponseAs<V> andThenJson(
+            Class<? extends V> clazz, ObjectMapper objectMapper, Predicate<AggregatedHttpResponse> predicate) {
+        return (BlockingConditionalResponseAs<V>) andThen(
+                AggregatedResponseAs.json(clazz, objectMapper, predicate), predicate);
+    }
+
+    /**
+     * Invokes {@link ConditionalResponseAs#andThen(ResponseAs, Predicate)} to add the mapping of
+     * {@link ResponseAs} and {@link Predicate}.
+     */
+    public BlockingConditionalResponseAs<V> andThenJson(
+            TypeReference<? extends V> typeRef, Predicate<AggregatedHttpResponse> predicate) {
+        return (BlockingConditionalResponseAs<V>) andThen(
+                AggregatedResponseAs.json(typeRef, predicate), predicate);
+    }
+
+    /**
+     * Invokes {@link ConditionalResponseAs#andThen(ResponseAs, Predicate)} to add the mapping of
+     * {@link ResponseAs} and {@link Predicate}.
+     */
+    public BlockingConditionalResponseAs<V> andThenJson(
+            TypeReference<? extends V> typeRef, ObjectMapper objectMapper,
+            Predicate<AggregatedHttpResponse> predicate) {
+        return (BlockingConditionalResponseAs<V>) andThen(
+                AggregatedResponseAs.json(typeRef, objectMapper, predicate), predicate);
+    }
+
+    /**
+     * Invokes {@link ConditionalResponseAs#orElse(ResponseAs)} and returns {@link ResponseAs} whose
+     * {@link Predicate} is evaluated as true.
+     */
+    public ResponseAs<HttpResponse, ResponseEntity<V>> orElseJson(Class<? extends V> clazz) {
+        return orElse(AggregatedResponseAs.json(clazz, SUCCESS_PREDICATE));
```

Comment:
Since we're already in a conditional, I think it's fine to just try and parse without checking if the response is successful or not. What do you think of just inputting an always `true` predicate?

```suggestion
        return orElse(AggregatedResponseAs.json(clazz,() -> true));
```

Reply:
It makes sense more! Thanks!
Users might use `orElse` to map json response which isn't Success.